### ISSUE TITLE
Basic Tutorial - error when debug initiates

### ISSUE THREAD

**Issue by AdamOranchak** [ProjectPhysX/FluidX3D#80]
C++ isn't my first language (Jupyter/Python) so this may be a NOOB issue. 

I'm getting: "Unhandled exception at 0x0000000000000000 in FluidX3D.exe: 0xC0000005: Access violation executing location " from "cl_int getDevices" Any ideas?

![Tutorial Error 2023-05-16 165424](https://github.com/ProjectPhysX/FluidX3D/assets/45951432/1a98181e-8d40-4e60-9550-9c87dae6848d)

**Comment by ProjectPhysX** [comment]
Hi @AdamOranchak,

try setting the compiler from `Debug` to `Release`.
Please let me know if this fixes the problem!

Regards,
Moritz

![image](https://github.com/ProjectPhysX/FluidX3D/assets/90851087/b1c03ece-a37f-4948-af04-e50dbdd3ea49)

**Comment by AdamOranchak** [comment]
Thank you for your timely response. Unfortunately, your suggestion didn't work for me. I've been digging into this but have found a solution yet.

## What I know so far:

- Alienware X17R laptop
- Windows 11 Home
- 11th i7-11800H @ 2.30GHz, 32.0 GB (31.7 GB usable)
- NVidia RTX 3070 Laptop GPU, Driver version 31.0.15.3179
- OpenCL.dll is locates in C:|WINDOWS\System32
- Intel UHD Graphics, Driver version 31.0.101.4369
- This driver appears to be referencing the NVidia OpenCL.dll

I found an [excellent lecture series by Professor Jan Verschelde](http://homepages.math.uic.edu/~jan/mcs572f16/mcs572notes/lec29.html) that provides a "sanity check" Python script that utilizes PyOpenCL. So far I haven't been able to get it running properly.

Clearly, something is preventing OpenCL from running properly. I don't think that this is a problem with your code.

**Comment by ProjectPhysX** [comment]
Can you run the FluidX3D benchmark from [here](https://github.com/ProjectPhysX/FluidX3D/releases/download/v2.6/FluidX3D-Benchmark-FP32-FP32-Windows.exe)? This will show you if the OpenCL devices are detected correctly.

**Comment by AdamOranchak** [comment]
Negative, Dr. I get the following splash screen for a second or 2 and then nothing. I checked the running processes and I'm confident that it is no longer in memory,
![vlcsnap-2023-05-20-18h31m51s895](https://github.com/ProjectPhysX/FluidX3D/assets/45951432/7d90d9dc-8854-49ae-a156-9c342dad9e23)

**Comment by ProjectPhysX** [comment]
This is a problem with your driver installation. Can you re-install the latest Nvidia drivers?

**Comment by AdamOranchak** [comment]
SOLVED! I posted an inquiry on NVidia's Developer Forum and received a suggestion to disable the Intel Graphic Adapter from the display adapter section of Windows Device Manager. This worked. Here are the results from the FluidX3D Benchmark:
![SOLVED - FluidX3D Benchmark -  2023-05-21](https://github.com/ProjectPhysX/FluidX3D/assets/45951432/9cb9517a-e53f-463b-94a0-3c76fff58b2b)

Here are some additional notes that may prove useful:

Prior to arriving at the solution, updated NVidia's CUDA Tool Kit. Initially, the installation reported that there wasn't a compatible version of Visual Studio on my machine. I performed a clean installation of Visual Studio Community 2022 and re-ran the tool kit installation successfully. I followed the[ CUDA Installation Guide for Windows](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html) to the letter with one minor deviation. I was able to compile and run deviceQuery, bandwidthTest, and Smoke Particles. (particles code is no longer on their repository)

However, even after those successes, FluidX3D benchmark didn't run and FluidX3D code wouldn't compile. Disabling the Intel driver fixed everything.

Thank you so very much for writing, sharing, and helping with your code which is an absolute thing of beauty!
